File: particle.py
from vector import Vector2

class Particle:
    def __init__(self, mass=1, pos=Vector2(0,0), vel=Vector2(0,0), color=(1,1,1)):
        self.mass = mass
        self.pos = pos
        self.vel = vel
        self.color = color
    
    def update(self, force, dt):
        acceleration = force / self.mass
        self.vel += acceleration * dt
        self.pos += self.vel * dt

File: src/simulation/particle.rs
use super::*;
use serde::{Serialize, Deserialize};

#[derive(Clone, Serialize, Deserialize)]
pub struct Particle {
    pub mass: f64,
    pub position: Vector2,
    pub velocity: Vector2,
    pub color: (f64, f64, f64),
}

impl Particle {
    pub fn update(&mut self, delta_time: f64, force: Vector2) {
        let acceleration = force / self.mass;
        self.velocity += acceleration * delta_time;
        self.position += self.velocity * delta_time;
    }
}
